File: lab0600/Scala/Main.scala
object Main {
  def main(args: Array[String]): Unit = {
    println("=== Scala Функциональное Программирование ===")
    
    // Базовые операции
    println("\n--- Базовые операции ---")
    BasicScala.main(Array())
    
    // Коллекции
    println("\n--- Коллекции ---")
    Collections.demonstrateCollections()
    
    // Обработка ошибок
    println("\n--- Обработка ошибок ---")
    ErrorHandling.demonstrateErrorHandling()
    
    // Pattern matching
    println("\n--- Pattern Matching ---")
    PatternMatching.demonstratePatternMatching()
    
    // Spark (требует установленного Spark)
    // SparkExample.main(Array())
  }
}

File: lab0600/Rust/src/main.rs
mod ownership;
mod iterators_closures;
mod pattern_matching;
mod error_handling;
mod comparison;

fn main() {
    println!("=== Rust Функциональное Программирование ===\n");
    
    // Базовые операции и система владения
    ownership::main();
    
    // Итераторы и замыкания
    iterators_closures::demonstrate_iterators();
    iterators_closures::main();
    
    // Pattern matching
    pattern_matching::demonstrate_pattern_matching();
    
    // Обработка ошибок
    error_handling::demonstrate_error_handling();
    error_handling::main();
    
    // Сравнительный анализ
    println!("\n=== Сравнительный анализ ===");
    comparison::main();
}

